### PR TITLE
Protocol to sweep CZ amplitude

### PR DESCRIPTION
**Pull request by andrea-pasquale**
I added an experiment which consists in sweeping the amplitude of the CZ flux pulse to compute both CZ angle and leakage. By selecting the angle closer to $\pi$.

Here is a possible runcard:
```yaml
- id: cz_sweep
  operation: cz_sweep
  targets: [[D1,D2]]
  parameters:
    flux_pulse_amplitude_min: 0.1995
    flux_pulse_amplitude_max: 0.2000
    flux_pulse_amplitude_step: 0.0001
    duration_max: 73
    duration_min: 70
    duration_step: 1
    theta_start: 0
    theta_end: 7
    theta_step: 0.5
    relaxation_time: 50_000
```

Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.
- [ ] Compatibility with Qibo modules (Please edit this section if the current pull request is not compatible with the following branches).
    - [ ] Qibo: `master`
    - [ ] Qibolab: `main`
    - [ ] Qibolab_platforms_qrc: `main`

### CODE REVIEW

**Comment by andrea-pasquale** [comment]
I've updated the code (it is still quite a bit messy) but the it seems to be working properly.

http://login.qrccluster.com:9000/eV94CZPnQReFKNeyvq7Rxg==/

I'm already putting it as ready to review but I might try to simply the code later on.

**Review by igres26 — APPROVED** [review]
Works alright for me.

File: src/qibocal/protocols/two_qubit_interaction/cz_sweep.py [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+                        np.where(
+                            np.logical_and(
+                                data[pair][target, control, setup].amp == amplitude,
+                                data[pair][target, control, setup].duration == duration,
+                            )
+                        )
+                    ]
+                    pguess = [
+                        np.max(target_data) - np.min(target_data),
+                        np.mean(target_data),
```
Comment (andrea-pasquale) [inline src/qibocal/protocols/two_qubit_interaction/cz_sweep.py:274]: Reduce code duplication in fitting

Comment (andrea-pasquale) [reply src/qibocal/protocols/two_qubit_interaction/cz_sweep.py:274]: I think the only problem is that the fit a bit messy, however given that people are already using it I would merge. I've opened an issue about this #962

**Review by Jacfomg — APPROVED** [review]
👍

**Comment by andrea-pasquale** [comment]
It could be nice to generalize it for the iSWAP gate if needed

**Pull request merged**